apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -171,76 +212,148 @@ static int create_file(char *path, ssize_t bufsize, file_type_t type)
 	return ret;
 }
 
-static int verify_random_file(char *path, int bufsize, int *index)
+static int verify_random_file_data(int fd, int bufsize)
+{
+	int len;
+	uint16_t crc;
+	int data_len = bufsize - sizeof(g_buf.crc);
+
+	len = read(fd, (void *)&g_buf, bufsize);
+	if (len != bufsize) {
+		/* If write is not finished properly */
+		printf("verify_random_file : datachunk is not written fully, ret : %d\n", len);
+		return ERROR;
+	} else {
+		/* Maybe this won't happen but check crc for double confirmation */
+		crc = crc16((const uint8_t *)g_buf.data, data_len);
+		if (crc != g_buf.crc) {
+			printf("verify_random_file : crc mismatch, calulcated crc : %d, written crc : %d\n", crc, g_buf.crc);
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+static int verify_text_file_data(int fd, char *sample_buf, uint16_t sample_crc, int bufsize)
+{
+	int len;
+	uint16_t crc;
+
+	len = read(fd, (void *)sample_buf, bufsize);
+	if (len != bufsize) {
+		/* If write is not finished properly */
+		printf("verify_text_file : datachunk is not written fully, ret : %d\n", len);
+		return ERROR;
+	} else {
+		/* Maybe this won't happen but check crc for double confirmation */
+		crc = crc16((const uint8_t *)sample_buf, bufsize);
+		if (crc != sample_crc) {
+			printf("verify_random_file : crc mismatch, calulcated crc : %d, Sample File crc : %d\n", crc, sample_crc);
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+static int verify_file(char *path, int bufsize, int *index, file_type_t type)
 {
 	int fd;
+	int fd_sample;
```

Comment:
Simply, fd1, fd2 is enough. but choice is yours.

Reply:
We'll keep it same if it's not causing problems?
fd1 and fd2 are kept for main file and backup file respectively.

So fd_sample specifically says this is the descriptor for sample text file.

Reply:
that is very common usage style of fd, vfs also using like that.
But anyway, I don't care about this. you can keep yours.